spec/models/grda_warehouse/tasks/identify_duplicates_spec.rb

```diff
@@ -7,123 +7,196 @@
+    it 'recognizes an obvious match' do
+      expect(check_for_obvious_match(client_in_source)).to be client_in_destination.id
     end
 
-    it 'merged client and client 2 into a single destination client' do
-      expect(client_in_source.destination_client.id).to eq(client_two_in_source.destination_client.id)
-      expect(GrdaWarehouse::Hud::Client.destination.count).to eq(1)
+    describe 'obvious match processing with a split' do
+      let!(:split) { create :grda_warehouse_client_split_history, split_from: client_in_source.id, split_into: client_in_destination.id }
+
+      it 'does not return an obvious match if it was split' do
+        expect(check_for_obvious_match(client_in_source)).to be_nil
+      end
     end
 
-    describe 'merge processing after a split' do
-      let(:user) { create :user }
+    describe 'find merge candidates' do
+      let!(:client_two_in_source) { create :grda_warehouse_hud_client, data_source: source_data_source }
+      let!(:warehouse_data_source) { create :destination_data_source }
 
       before(:each) do
-        destination_client = client_in_source.destination_client
-        split_client_id = destination_client.source_clients.last.id
-        destination_client.split(
-          [split_client_id],
-          destination_client.id,
-          destination_client.id,
-          user,
-        )
+        processor = GrdaWarehouse::Tasks::IdentifyDuplicates.new(run_post_processing: false)
+        processor.run!
+        processor.match_existing!
       end
 
-      it 'sees two destination clients after split' do
-        expect(GrdaWarehouse::Hud::Client.destination.count).to eq(2)
+      it 'merged client and client 2 into a single destination client' do
+        expect(client_in_source.destination_client.id).to eq(client_two_in_source.destination_client.id)
+        expect(GrdaWarehouse::Hud::Client.destination.count).to eq(1)
       end
 
-      it 'does not re-merge the split clients' do
-        GrdaWarehouse::Tasks::IdentifyDuplicates.new(run_post_processing: false).match_existing!
-
-        expect(GrdaWarehouse::Hud::Client.destination.count).to eq(2)
+      describe 'merge processing after a split' do
+        let(:user) { create :user }
+
+        before(:each) do
+          destination_client = client_in_source.destination_client
+          split_client_id = destination_client.source_clients.last.id
+          destination_client.split(
+            [split_client_id],
+            destination_client.id,
+            destination_client.id,
+            user,
+          )
+        end
+
+        it 'sees two destination clients after split' do
+          expect(GrdaWarehouse::Hud::Client.destination.count).to eq(2)
+        end
+
+        it 'does not re-merge the split clients' do
+          GrdaWarehouse::Tasks::IdentifyDuplicates.new(run_post_processing: false).match_existing!
+
+          expect(GrdaWarehouse::Hud::Client.destination.count).to eq(2)
+        end
```

Comment:
This is fine; it looks like we have the pattern in these tests of processing in a before block. Maybe it's too much trouble but perhaps it would be more complete to move the processing in an expect block to verify the values are actually changing before and after the split?

```suggestion
        def split_client
          destination_client = client_in_source.destination_client
          split_client_id = destination_client.source_clients.last.id
          destination_client.split(
            [split_client_id],
            destination_client.id,
            destination_client.id,
            user,
          )
        end
        
        let(:destination_scope) { GrdaWarehouse::Hud::Client.destination }

        it 'sees two destination clients after split' do
          expect { split_client }.to change(destination_scope, :count).from(1).to(2)
        end
        
        it 'does not re-merge the split clients' do
          expect do
            split_client
            GrdaWarehouse::Tasks::IdentifyDuplicates.new(run_post_processing: false).match_existing!
          end.to change(destination_scope, :count).from(1).to(2)
        end
```